executor/memtable_reader_test.go

```diff
@@ -941,3 +944,706 @@ func (s *testMemTableReaderSuite) TestTiDBClusterLogError(c *C) {
+		}
+	}
+	tk := testkit.NewTestKit(c, s.store)
+	for _, cas := range cases {
+		sql := "select * from information_schema.tidb_hot_regions_history"
+		if len(cas.conditions) > 0 {
+			sql = fmt.Sprintf("%s where %s", sql, strings.Join(cas.conditions, " and "))
+		}
+		result := tk.MustQuery(sql)
+		warnings := tk.Se.GetSessionVars().StmtCtx.GetWarnings()
+		c.Assert(len(warnings), Equals, 0, Commentf("unexpected warnigns: %+v", warnings))
+		var expected []string
+		for _, row := range cas.expected {
+			expectedRow := row
+			expected = append(expected, strings.Join(expectedRow, " "))
+		}
+		result.Check(testkit.Rows(expected...))
+	}
+}
+
+func (s *testHotRegionsHistoryTableSuite) TestTiDBHotRegionsHistoryError(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	fpName := "github.com/pingcap/tidb/executor/mockTiDBHotRegionsHistory"
+	c.Assert(failpoint.Enable(fpName, `return("")`), IsNil)
+	defer func() { c.Assert(failpoint.Disable(fpName), IsNil) }()
+
+	// Test without start time error
+	rs, err := tk.Exec("select * from information_schema.tidb_hot_regions_history")
+	c.Assert(err, IsNil)
+	_, err = session.ResultSetToStringSlice(context.Background(), tk.Se, rs)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "denied to scan hot regions, please specified the start time, such as `update_time > '2020-01-01 00:00:00'`")
+	c.Assert(rs.Close(), IsNil)
+
+	// Test without end time error.
+	rs, err = tk.Exec("select * from information_schema.tidb_hot_regions_history where update_time>='2019/08/26 06:18:13.011'")
+	c.Assert(err, IsNil)
+	_, err = session.ResultSetToStringSlice(context.Background(), tk.Se, rs)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "denied to scan hot regions, please specified the end time, such as `update_time < '2020-01-01 00:00:00'`")
+	c.Assert(rs.Close(), IsNil)
+}
+
+var regions []helper.RegionInfo
+
+func tikvRegionStatusHandler(w http.ResponseWriter, r *http.Request) {
+	start_key := r.URL.Query().Get("start_key")
+	end_key := r.URL.Query().Get("end_key")
+	start := string(start_key)
+	end := string(end_key)
+	regionsInfo := helper.RegionsInfo{
+		Count:   0,
+		Regions: make([]helper.RegionInfo, 0),
+	}
+	for _, region := range regions {
+		if start >= region.StartKey &&
+			end <= region.EndKey {
+			regionsInfo.Regions = append(regionsInfo.Regions, region)
+		}
+	}
+	jsonResp, err := json.Marshal(regionsInfo)
+	if err != nil {
+		writeJSONError(w, http.StatusInternalServerError, "unable to marshal resp", err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonResp)
+}
+
+var _ = SerialSuites(&testTikvRegionStatusTableSuite{testInfoschemaTableSuiteBase: &testInfoschemaTableSuiteBase{}})
+
+type testTikvRegionStatusTableSuite struct {
+	*testInfoschemaTableSuiteBase
+	httpServers []*httptest.Server
+	startTime   time.Time
+}
+
+func (s *testTikvRegionStatusTableSuite) SetUpSuite(c *C) {
+	s.testInfoschemaTableSuiteBase.SetUpSuite(c)
+	store := &mockStoreWithMultiPD{
```

Comment:
how about use `mockStore` in `infoschem_reade_test.go`, because there is only one pd server here.